libfwupdplugin/fu-bytes.c

```diff
@@ -68,16 +68,33 @@ fu_bytes_set_contents(const gchar *filename, GBytes *bytes, GError **error)
 GBytes *
 fu_bytes_get_contents(const gchar *filename, GError **error)
 {
-	gchar *data = NULL;
-	gsize len = 0;
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(GMappedFile) mapped_file = NULL;
 
 	g_return_val_if_fail(filename != NULL, NULL);
 	g_return_val_if_fail(error == NULL || *error == NULL, NULL);
 
-	if (!g_file_get_contents(filename, &data, &len, error))
-		return NULL;
-	g_debug("reading %s with %" G_GSIZE_FORMAT " bytes", filename, len);
-	return g_bytes_new_take(data, len);
+	/* try as a mapped file, falling back to reading it as a blob instead */
+	mapped_file = g_mapped_file_new(filename, FALSE, &error_local);
+	if (mapped_file == NULL) {
+		gchar *data = NULL;
+		gsize len = 0;
+		if (!g_file_get_contents(filename, &data, &len, error))
+			return NULL;
+		g_debug("failed to read as mapped file, so reading %s of size 0x%x: %s",
```

Comment:
shouldn't you be using `G_GSIZE_FORMAT`?

Reply:
do you know how to coerce G_GSIZE_FORMAT to display in hex format?

Reply:
Doh, it's easier than I thought. Fixed.